### ISSUE TITLE
[Qt6, pyside6] QAction has no attribute `menu` in fresh napari install

### ISSUE THREAD

**Issue by kephale** [napari/napari#6135]
## 🐛 Bug

I ran into a problem recently setting up a fresh napari environment, and it looks like someone else did recently (https://github.com/DeepLabCut/DeepLabCut/issues/2344). 

```
>>> import napari
>>> v = napari.Viewer()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/kharrington/git/kephale/nesoi/repos/napari/napari/viewer.py", line 67, in __init__
    self._window = Window(self, show=show)
  File "/Users/kharrington/git/kephale/nesoi/repos/napari/napari/_qt/qt_main_window.py", line 619, in __init__
    self._add_menus()
  File "/Users/kharrington/git/kephale/nesoi/repos/napari/napari/_qt/qt_main_window.py", line 775, in _add_menus
    self.file_menu = menus.FileMenu(self)
  File "/Users/kharrington/git/kephale/nesoi/repos/napari/napari/_qt/menus/file_menu.py", line 149, in __init__
    self._rebuild_samples_menu()
  File "/Users/kharrington/git/kephale/nesoi/repos/napari/napari/_qt/menus/file_menu.py", line 213, in _rebuild_samples_menu
    ).menu()
AttributeError: 'QAction' object has no attribute 'menu'
```

It looks like QAction.menu has been deprecated (https://forums.ankiweb.net/t/add-on-issues-with-anki-2-1-50-beta/15631, https://forum.qt.io/topic/134911/qaction-setmenu-is-deprecated-what-is-the-recommended-alternative)

## To Reproduce

System: MacOS 13.5 arm64

```
mamba create -y -n napari-poor-mans-octree -c conda-forge python=3.10 pip
mamba activate napari-poor-mans-octree
pip install -e ".[dev,testing,pyside6_experimental]"
```

## Environment

```
> conda list                                                                                                                       ~/nesoi/repos/napari
# packages in environment at /Users/kharrington/micromamba/envs/napari-poor-mans-octree:
#
# Name                    Version                   Build  Channel
alabaster                 0.7.13                   pypi_0    pypi
app-model                 0.2.0                    pypi_0    pypi
appdirs                   1.4.4                    pypi_0    pypi
appnope                   0.1.3                    pypi_0    pypi
asciitree                 0.3.3                    pypi_0    pypi
asttokens                 2.2.1                    pypi_0    pypi
attrs                     23.1.0                   pypi_0    pypi
babel                     2.12.1                   pypi_0    pypi
backcall                  0.2.0                    pypi_0    pypi
black                     23.7.0                   pypi_0    pypi
build                     0.10.0                   pypi_0    pypi
bzip2                     1.0.8                h3422bc3_4    conda-forge
ca-certificates           2023.7.22            hf0a4a13_0    conda-forge
cachey                    0.2.1                    pypi_0    pypi
certifi                   2023.7.22                pypi_0    pypi
cfgv                      3.3.1                    pypi_0    pypi
charset-normalizer        3.2.0                    pypi_0    pypi
check-manifest            0.49                     pypi_0    pypi
click                     8.1.6                    pypi_0    pypi
cloudpickle               2.2.1                    pypi_0    pypi
comm                      0.1.4                    pypi_0    pypi
contourpy                 1.1.0                    pypi_0    pypi
coverage                  7.2.7                    pypi_0    pypi
cycler                    0.11.0                   pypi_0    pypi
dask                      2023.8.0                 pypi_0    pypi
debugpy                   1.6.7.post1              pypi_0    pypi
decorator                 5.1.1                    pypi_0    pypi
docstring-parser          0.15                     pypi_0    pypi
docutils                  0.17.1                   pypi_0    pypi
entrypoints               0.4                      pypi_0    pypi
exceptiongroup            1.1.2                    pypi_0    pypi
executing                 1.2.0                    pypi_0    pypi
fasteners                 0.18                     pypi_0    pypi
filelock                  3.12.2                   pypi_0    pypi
fonttools                 4.42.0                   pypi_0    pypi
freetype-py               2.4.0                    pypi_0    pypi
fsspec                    2023.6.0                 pypi_0    pypi
heapdict                  1.0.1                    pypi_0    pypi
hsluv                     5.0.3                    pypi_0    pypi
hypothesis                6.82.3                   pypi_0    pypi
identify                  2.5.26                   pypi_0    pypi
idna                      3.4                      pypi_0    pypi
imageio                   2.31.1                   pypi_0    pypi
imagesize                 1.4.1                    pypi_0    pypi
importlib-metadata        6.8.0                    pypi_0    pypi
in-n-out                  0.1.8                    pypi_0    pypi
iniconfig                 2.0.0                    pypi_0    pypi
ipykernel                 6.25.1                   pypi_0    pypi
ipython                   8.14.0                   pypi_0    pypi
ipython-genutils          0.2.0                    pypi_0    pypi
jedi                      0.19.0                   pypi_0    pypi
jinja2                    3.1.2                    pypi_0    pypi
jsonschema                4.19.0                   pypi_0    pypi
jsonschema-specifications 2023.7.1                 pypi_0    pypi
jupyter-client            8.3.0                    pypi_0    pypi
jupyter-core              5.3.1                    pypi_0    pypi
kiwisolver                1.4.4                    pypi_0    pypi
lazy-loader               0.3                      pypi_0    pypi
libffi                    3.4.2                h3422bc3_5    conda-forge
libsqlite                 3.42.0               hb31c410_0    conda-forge
libzlib                   1.2.13               h53f4e23_5    conda-forge
llvmlite                  0.40.1                   pypi_0    pypi
locket                    1.0.0                    pypi_0    pypi
magicgui                  0.7.2                    pypi_0    pypi
markdown-it-py            3.0.0                    pypi_0    pypi
markupsafe                2.1.3                    pypi_0    pypi
matplotlib                3.7.2                    pypi_0    pypi
matplotlib-inline         0.1.6                    pypi_0    pypi
mdurl                     0.1.2                    pypi_0    pypi
mpmath                    1.3.0                    pypi_0    pypi
mypy-extensions           1.0.0                    pypi_0    pypi
napari                    0.5.0a2.dev384+g2e1d7c12          pypi_0    pypi
napari-console            0.0.8                    pypi_0    pypi
napari-plugin-engine      0.2.0                    pypi_0    pypi
napari-plugin-manager     0.1.0a2                  pypi_0    pypi
napari-svg                0.1.10                   pypi_0    pypi
ncurses                   6.4                  h7ea286d_0    conda-forge
nest-asyncio              1.5.7                    pypi_0    pypi
networkx                  3.1                      pypi_0    pypi
nodeenv                   1.8.0                    pypi_0    pypi
npe2                      0.7.2                    pypi_0    pypi
numba                     0.57.1                   pypi_0    pypi
numcodecs                 0.11.0                   pypi_0    pypi
numpy                     1.24.4                   pypi_0    pypi
numpydoc                  1.5.0                    pypi_0    pypi
openssl                   3.1.2                h53f4e23_0    conda-forge
packaging                 23.1                     pypi_0    pypi
pandas                    2.0.3                    pypi_0    pypi
parso                     0.8.3                    pypi_0    pypi
partd                     1.4.0                    pypi_0    pypi
pathspec                  0.11.2                   pypi_0    pypi
pexpect                   4.8.0                    pypi_0    pypi
pickleshare               0.7.5                    pypi_0    pypi
pillow                    10.0.0                   pypi_0    pypi
pint                      0.22                     pypi_0    pypi
pip                       23.2.1             pyhd8ed1ab_0    conda-forge
platformdirs              3.10.0                   pypi_0    pypi
pluggy                    1.2.0                    pypi_0    pypi
pooch                     1.7.0                    pypi_0    pypi
pre-commit                3.3.3                    pypi_0    pypi
prompt-toolkit            3.0.39                   pypi_0    pypi
psutil                    5.9.5                    pypi_0    pypi
psygnal                   0.9.1                    pypi_0    pypi
ptyprocess                0.7.0                    pypi_0    pypi
pure-eval                 0.2.2                    pypi_0    pypi
pydantic                  1.10.12                  pypi_0    pypi
pyopengl                  3.1.7                    pypi_0    pypi
pyparsing                 3.0.9                    pypi_0    pypi
pyproject-hooks           1.0.0                    pypi_0    pypi
pyside6                   6.5.2                    pypi_0    pypi
pyside6-addons            6.5.2                    pypi_0    pypi
pyside6-essentials        6.5.2                    pypi_0    pypi
pytest                    7.4.0                    pypi_0    pypi
pytest-cov                4.1.0                    pypi_0    pypi
pytest-pretty             1.2.0                    pypi_0    pypi
pytest-qt                 4.2.0                    pypi_0    pypi
python                    3.10.12         h01493a6_0_cpython    conda-forge
python-dateutil           2.8.2                    pypi_0    pypi
python-dotenv             1.0.0                    pypi_0    pypi
pytz                      2023.3                   pypi_0    pypi
pywavelets                1.4.1                    pypi_0    pypi
pyyaml                    6.0.1                    pypi_0    pypi
pyzmq                     25.1.1                   pypi_0    pypi
qtconsole                 5.4.3                    pypi_0    pypi
qtpy                      2.3.1                    pypi_0    pypi
readline                  8.2                  h92ec313_1    conda-forge
referencing               0.30.2                   pypi_0    pypi
requests                  2.31.0                   pypi_0    pypi
rich                      13.5.2                   pypi_0    pypi
rpds-py                   0.9.2                    pypi_0    pypi
scikit-image              0.21.0                   pypi_0    pypi
scipy                     1.11.1                   pypi_0    pypi
setuptools                68.0.0             pyhd8ed1ab_0    conda-forge
shiboken6                 6.5.2                    pypi_0    pypi
snowballstemmer           2.2.0                    pypi_0    pypi
sortedcontainers          2.4.0                    pypi_0    pypi
sphinx                    4.5.0                    pypi_0    pypi
sphinxcontrib-applehelp   1.0.4                    pypi_0    pypi
sphinxcontrib-devhelp     1.0.2                    pypi_0    pypi
sphinxcontrib-htmlhelp    2.0.1                    pypi_0    pypi
sphinxcontrib-jsmath      1.0.1                    pypi_0    pypi
sphinxcontrib-qthelp      1.0.3                    pypi_0    pypi
sphinxcontrib-serializinghtml 1.1.5                    pypi_0    pypi
stack-data                0.6.2                    pypi_0    pypi
superqt                   0.5.0                    pypi_0    pypi
sympy                     1.12                     pypi_0    pypi
tensorstore               0.1.41                   pypi_0    pypi
tifffile                  2023.7.18                pypi_0    pypi
tk                        8.6.12               he1e0b03_0    conda-forge
tomli                     2.0.1                    pypi_0    pypi
tomli-w                   1.0.0                    pypi_0    pypi
toolz                     0.12.0                   pypi_0    pypi
torch                     2.0.1                    pypi_0    pypi
tornado                   6.3.2                    pypi_0    pypi
tqdm                      4.66.1                   pypi_0    pypi
traitlets                 5.9.0                    pypi_0    pypi
typer                     0.9.0                    pypi_0    pypi
typing-extensions         4.7.1                    pypi_0    pypi
tzdata                    2023.3                   pypi_0    pypi
urllib3                   2.0.4                    pypi_0    pypi
vispy                     0.13.0                   pypi_0    pypi
wcwidth                   0.2.6                    pypi_0    pypi
wheel                     0.41.1             pyhd8ed1ab_0    conda-forge
wrapt                     1.15.0                   pypi_0    pypi
xarray                    2023.7.0                 pypi_0    pypi
xz                        5.2.6                h57fd34a_0    conda-forge
zarr                      2.16.0                   pypi_0    pypi
zipp                      3.16.2                   pypi_0    pypi
```

**Comment by psobolewskiPhD** [comment]
Seems pyside6 specific? Not hitting this on pyqt6 -- installed same as your instructions just pyqt6.
Both are experimental, but Pyside6 has more issues at the moment.

**Comment by kephale** [comment]
@psobolewskiPhD great! Thank you. I had extra issues due to some clobbering issue with a micromamba install that automatically ran shell init commands (which was causing issues with pyqt as well). After cleaning up everything and starting from scratch I'm fine with pyqt as well.

That said DeepLabCut requires pyside6 so this still a problem for this DLC issue (https://github.com/DeepLabCut/DeepLabCut/issues/2344)

**Comment by dalthviz** [comment]
Hi, just in case, checking the related ticket (https://bugreports.qt.io/browse/PYSIDE-1627) seems like this was an issue for PySide 6 6.1.2 and [a fix was done for that](https://codereview.qt-project.org/c/pyside/pyside-setup/+/360913). Also, checking the documentation, seems like QAction.menu [is not part of the deprecated members of `QAction`](https://doc.qt.io/qt-6/qaction-obsolete.html) and you can see [the docs for `menu` for example](https://doc.qt.io/qt-6/qaction.html#menu) 🤔 Could this be a PySide6 issue/regression?

**Comment by kephale** [comment]
Aha, nice 🕵️  work! 

So we should pin the version of PySide6? Would that be to `>=6.3.0`?

**Comment by psobolewskiPhD** [comment]
@kephale I can't reproduce this with your pyside6 instructions because `pyside6_experimental` pulls 6.5.2 (which is listed in your env) and then I hit this TypeError:
https://github.com/napari/napari/issues/5703

Using `pyside6==6.4.2` works, but has other severe issue:
https://github.com/napari/napari/issues/5644

The error you see does occur with pyqt6==6.3.1, but is resolved with pyqt6==6.4.0

**Comment by psobolewskiPhD** [comment]
I'm going to close, because the original issue is fixed upstream and the remaining pyside6 issues have their own issues.
So in the mean time the solution is to use PyQt6 for Qt6.
But if folks think there's something to go on here, feel free to re-open.
